child_compassion/models/compassion_hold.py

```diff
@@ -128,3 +131,20 @@ def hold_sent(self, vals):
                 # delete child if no hold_id received
                 child_to_update.unlink()
                 hold.unlink()
+
+    @api.model
+    def reservation_to_hold(self, commkit_data):
+        """ Called when a reservation gots converted to a hold. """
+        mapping = ReservationToHoldMapping(self.env)
+        hold_data = commkit_data.get(
+            'GlobalPartnerBeneficiaryReservationToHoldNotification')
+        child_global_id = hold_data and hold_data.get('Beneficiary_GlobalID')
+        if child_global_id:
+            child = self.env['compassion.child'].create(

```

Comment:
In fact it is the same because at the end I have to add `hold.child_id = child` with your solution.
It just inverse the order of creation between child and hold.